test/sqllogictest/with_mutually_recursive.slt

```diff
@@ -10,11 +10,13 @@
 mode cockroach
 
 ## Test a plausibly correct recursive query.
-statement error not supported in HIR
+query I
 WITH MUTUALLY RECURSIVE
     foo (a int, b int) AS (SELECT 1, 2 UNION SELECT a, 7 FROM bar),
     bar (a int) as (SELECT a FROM foo)
 SELECT * FROM bar;
+----
+1
```

Comment:
Here is my attempt to work out how this evaluation should work in practice:

```
i       foo                   bar
0       {}                    {}
1       {(1,2)}               {}
2       {(1,2)}               {(1)}
3       {(1,2),(1,7)}         {(1)}
4       {(1,2),(1,7)}         {(1),(1)}
5       {(1,2),(1,7),(1,7)}   {(1),(1)}
6       {(1,2),(1,7),(1,7)}   {(1),(1),(1)}
```

According to my interpretation of #16445 this will never converge.

Also, after 1 iteration (according to the counting above) the value of `SELECT * FROM bar` should be empty.

Reply:
I think it iteration 1 `bar` should contain `(1)`, because it selects out of `foo` which contains `(1,2)` at the time. The semantics are "sequential execution" where each binding sees the most recent contents of each other binding, in part so that we get the right behavior out of existing `Let` stacks (no delay before we can observe the contents of prior bindings).

I think it will converge, as `UNION` is implicitly followed by a distinct (rather than a `UNION ALL` which is not). So `foo` should converge to the distinct elements `{(1, 2), (1, 7)}`, and then `bar` from it. No thought was put in to determining if this is a worthwhile query to use as a test though; we should certainly revisit!

Reply:
I missed the sequential execution part and I guess I was too tired at 7AM to do the distinct. With corrections, it should be:

```text
i       foo                   bar
0       {}                    {}
1       {(1,2)}               {(1)}
2       {(1,2),(1,7)}         {(1),(1)}
3       {(1,2),(1,7)}         {(1),(1)}
```

so if we implement LIR lowering beyond the first iteration the final result should be `{(1),(1)}`.